packages/js/src/ui/context/CountContext.tsx

```diff
@@ -41,29 +33,28 @@ export const CountProvider = (props: CountProviderProps) => {
         return;
       }
 
-      setUnreadCounts((oldMap) => {
-        const newMap = new Map(oldMap);
-        newMap.set(defaultFilterKey, data.totalUnreadCount);
-
-        return newMap;
-      });
+      setTotalUnreadCount(data.totalUnreadCount);
     },
   });
 
   useWebSocketEvent({
     event: 'notifications.notification_received',
     eventHandler: async (data) => {
       const notification = data.result;
-      for (let i = 0; i < props.filters.length; i++) {
-        const filter = props.filters[i];
-        const filterKey = createKey(filter);
-        if (!areTagsEqual(filter.tags, notification.tags)) {
+      const allTabs = tabs();
+      for (let i = 0; i < allTabs.length; i++) {
+        const tab = allTabs[i];
+        const tags = tab.value;
+        const allNotifications = tags.length === 0;
+        const includeTags = notification.tags?.every((tag) => tags.includes(tag));
+        if (!allNotifications && !includeTags) {
```

Comment:
when the `tags: []` it means that we need to calculate all new messages count; but when it's not empty then we should compare if every notification tag is included in the tab tags